nav2_costmap_2d/CMakeLists.txt

```diff
@@ -26,17 +26,10 @@ find_package(visualization_msgs REQUIRED)
 find_package(angles REQUIRED)
 
 remove_definitions(-DDISABLE_LIBUSB-1.0)
-find_package(Eigen3 REQUIRED)
+find_package(Eigen3 3.3 REQUIRED)
```

Comment:
Why not use the installed / OS  version? This adds fragility that we have to keep updating this by the current ubuntu version.

Reply:
This specified a minimum version, not an exact match. Ubuntu 22 comes with Eigen 3.4, and the build passes, so this proves that it's not an exact match.

Reply:
How did you pick 3.3?

Reply:
I just tested. NAV2 compiles with Eigen3.3.9. That said, CI compiles with 3.4(what comes with Ubuntu 22), so 3.3 is untested. We can leave as-is.